Week3/homework/step2-6.js

```diff
@@ -4,13 +4,19 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 function printArray2d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  if (Array.isArray(arr)) {
+    arr.forEach(elem => printArray2d(elem));
+  } else {
+    console.log(arr);
+  }
 }
 
 function printArray3d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  if (Array.isArray(arr)) {
+    arr.forEach(elem => printArray3d(elem));
+  } else {
+    console.log(arr);
+  }
 }
```

Comment:
👍 for using a recursive function (that we did not explain in class) that supports arrays of any dimension. You are the first student to do this. 

However, I do see some code duplication here (not DRY). Can you change this code such that you still have two functions `printArray2d` and `printArray3d` (required to pass the unit tests), but with the duplicated code factored out into a third function?